src/rail_gui.cpp

```diff
@@ -1503,10 +1504,24 @@ struct BuildSignalWindow : public PickerWindowBase {
 				y + this->IsWidgetLowered(widget_index));
 	}
 
+	void SetSignalUIMode() {
+		this->show_non_path_signals = (_settings_client.gui.enable_signal_gui == 1);
+		this->GetWidget<NWidgetStacked>(WID_BS_SEMAPHORE_NORM_SEL)->SetDisplayedPlane(this->show_non_path_signals ? 0 : SZSP_NONE);
+		this->GetWidget<NWidgetStacked>(WID_BS_ELECTRIC_NORM_SEL)->SetDisplayedPlane(this->show_non_path_signals ? 0 : SZSP_NONE);
+		this->GetWidget<NWidgetStacked>(WID_BS_SEMAPHORE_ENTRY_SEL)->SetDisplayedPlane(this->show_non_path_signals ? 0 : SZSP_NONE);
+		this->GetWidget<NWidgetStacked>(WID_BS_ELECTRIC_ENTRY_SEL)->SetDisplayedPlane(this->show_non_path_signals ? 0 : SZSP_NONE);
+		this->GetWidget<NWidgetStacked>(WID_BS_SEMAPHORE_EXIT_SEL)->SetDisplayedPlane(this->show_non_path_signals ? 0 : SZSP_NONE);
+		this->GetWidget<NWidgetStacked>(WID_BS_ELECTRIC_EXIT_SEL)->SetDisplayedPlane(this->show_non_path_signals ? 0 : SZSP_NONE);
+		this->GetWidget<NWidgetStacked>(WID_BS_SEMAPHORE_COMBO_SEL)->SetDisplayedPlane(this->show_non_path_signals ? 0 : SZSP_NONE);
+		this->GetWidget<NWidgetStacked>(WID_BS_ELECTRIC_COMBO_SEL)->SetDisplayedPlane(this->show_non_path_signals ? 0 : SZSP_NONE);
```

Comment:
Does this really need to be 8 separate selection widgets, not just 2 (one for top row, one for bottom row)?

Reply:
If you do that, the NC_EQUALSIZE on the parent widget no longer usefully works, and you'd have to do the sizing manually.

Reply:
Thanks for answering this, @JGRennison.